presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkTaskRdd.java

```diff
@@ -67,7 +79,8 @@ public static <T extends PrestoSparkTaskOutput> PrestoSparkTaskRdd<T> create(
             Optional<PrestoSparkTaskSourceRdd> taskSourceRdd,
             // fragmentId -> RDD
             Map<String, RDD<Tuple2<MutablePartitionId, PrestoSparkMutableRow>>> shuffleInputRddMap,
-            PrestoSparkTaskProcessor<T> taskProcessor)
+            PrestoSparkTaskProcessor<T> taskProcessor,
+            boolean isNativeExecutionEnabled)
```

Comment:
I'm not super clear about the context. But I'm seeing there is some attempt trying to reuse and accommodate PrestoSparkTaskRdd for native execution. Adding a flag to the class and branching out in the codebase based on the flag looks more like a hack. We normally do this if we have no other options. Here a subclass is a cleaner solution. I'm going to make the change